File: Pickle.Web/Hubs/ChatHub.cs
using System.Threading.Tasks;
using Microsoft.AspNet.SignalR;
using Pickle.Web.Api.Providers;
using System;

namespace Pickle.Web.Hubs
{
    [Authorize]
    public class ChatHub : Hub<IChatHubClient>
    {
        private readonly IUsernameProvider usernameProvider;

        public ChatHub(IUsernameProvider usernameProvider)
        {
            this.usernameProvider = usernameProvider;
        }

        public override Task OnConnected()
        {
            return base.OnConnected();
        }

        public async void Send(string hubId, string channelId, string message, DateTime createdDate)
        {
            var username = await this.usernameProvider.GetUsername();

            Clients.All.BroadcastMessage(hubId, channelId, username, message, createdDate);
        }
    }
}

File: Pickle.Web/app/Hubs/ChatHub.ts
/// <reference path="../typings/signalr/signalr.d.ts" />

interface SignalR {
    chatHub: IChatHub;
}

interface IChatHub {
    client: ChatClient;
    server: ChatServer;
}

interface ChatClient {
    broadcastMessage(channelId: string, name: string, message: string): void;
}

interface ChatServer {
    send(channelId: string, message: string): JQueryPromise<void>;
}